app/routes/user_exercises.rb

```diff
@@ -50,11 +50,15 @@ class UserExercises < Core
       end
 
       post '/exercises/archive' do
-        exercises = current_user.exercises.find(params['exercise_ids'])
-        exercises.each(&:archive!)
-
-        flash[:success] = "Your exercises have been archived."
-        redirect "/#{current_user.username}"
+        begin
+          exercises = current_user.exercises.find(params['exercise_ids'])
+          exercises.each(&:archive!)
+          flash[:success] = "Your exercises have been archived."
+        rescue ActiveRecord::RecordNotFound => e
```

Comment:
Rather than rescuing a record not found, could we do a count on `exercises` and use a flash if the count is 0?